addons/binding/org.openhab.binding.facerecognition/src/main/java/org/openhab/binding/facerecognition/handler/FaceRecognitionHandler.java

```diff
@@ -0,0 +1,310 @@
+import org.bytedeco.javacpp.opencv_core.Rect;
+import org.bytedeco.javacpp.opencv_core.RectVector;
+import org.bytedeco.javacpp.opencv_face.FaceRecognizer;
+import org.bytedeco.javacpp.opencv_objdetect.CascadeClassifier;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.HttpResponseException;
+import org.eclipse.jetty.client.api.Authentication;
+import org.eclipse.jetty.client.api.AuthenticationStore;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.util.BasicAuthentication;
+import org.eclipse.jetty.client.util.DigestAuthentication;
+import org.eclipse.jetty.http.HttpStatus;
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.RawType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link FaceRecognitionHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Philipp Meisberger - Initial contribution
+ */
+public class FaceRecognitionHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(FaceRecognitionHandler.class);
+    private final ScheduledExecutorService service = Executors.newSingleThreadScheduledExecutor();
+    private final FaceRecognizer faceRecognizer = createLBPHFaceRecognizer();
+    private final HttpClient httpClient;
+    private final ChannelUID imageChannelUID;
+    private final ChannelUID accessGrantedImageChannelUID;
+    private final ChannelUID accessGrantedUsernameChannelUID;
+    private final ChannelUID accessGrantedScoreChannelUID;
+    private final ChannelUID accessGrantedTimeChannelUID;
+    private final ChannelUID accessDeniedImageChannelUID;
+    private final ChannelUID accessDeniedUsernameChannelUID;
+    private final ChannelUID accessDeniedScoreChannelUID;
+    private final ChannelUID accessDeniedTimeChannelUID;
+    private final static String MODELS_FILE = "/var/lib/openhab2/etc/org.openhab.binding.facerecognition.models.xml";
+    private final static String FACE_CLASSIFIER_FILE = "/usr/share/opencv/haarcascades/haarcascade_frontalface_default.xml";
+    private ScheduledFuture<?> job;
+    private CascadeClassifier faceCascade;
+    private Mat accessDeniedFace;
+    private BigDecimal interval;
+    private BigDecimal threshold;
+
+    public FaceRecognitionHandler(Thing thing) {
+        super(thing);
+        httpClient = new HttpClient();
```

Comment:
Use the http client factory service from `org.eclipse.smarthome.core.net`.
You need to acquire that service in your factory and hand it to your thing handler as a parameter